app/models/middleware_server.rb

```diff
@@ -5,4 +5,19 @@ class MiddlewareServer < ActiveRecord::Base
   has_many :middleware_deployments, :foreign_key => "server_id", :dependent => :destroy
   serialize :properties
   acts_as_miq_taggable
+
+  # Callback from _single_quad.html.haml to determine the icon
+  # we want to display a different icon depending of the type
+  # of server we have.
+  # See also MiddlewareServerController#listicon_image for the counterpart for lists.
+  def item_image
```

Comment:
To me having it as is makes some sense as the model has the full information and can provide it if it wants.

Also looking at `_single_quad.html.haml` I guess switch-case like

```
  - elsif item.kind_of?(ManageIQ::Providers::Foreman::ConfigurationManager) || item.kind_of?(ManageIQ::Providers::AnsibleTower::ConfigurationManager)
    - img = "vendor-#{item.image_name}"
```

could be omitted going forward as the provider plugin would deliver the information.

But of course that is just an idea.